crates/fluvio-channel/src/lib.rs

```diff
@@ -345,7 +345,8 @@ impl ChannelConfig {
             toml::to_vec(self).map_err(|err| IoError::new(ErrorKind::Other, format!("{}", err)))?;
 
         let mut file = File::create(path_ref)?;
-        file.write_all(&toml)
+        file.write_all(&toml)?;
+        file.sync_all()
```

Comment:
Since this is only important in windows, I thin that we could use a feature flag here:
```#[cfg(windows)]```

We could use https://github.com/alexcrichton/cfg-if

The same applies to all the changes in the first commit